scanpy/preprocessing/_highly_variable_genes.py

```diff
@@ -506,11 +512,12 @@ def highly_variable_genes(
                 na_position='last',
                 inplace=True,
             )
-            df['highly_variable'] = False
-            df.highly_variable.iloc[:n_top_genes] = True
-            df = df.loc[adata.var_names]
+            high_var = np.zeros(df.shape[0])
+            high_var[:n_top_genes] = True
+            df['highly_variable'] = high_var.astype(bool)
+            df = df.loc[adata.var_names, :]
```

Comment:
These were needed to get rid of a `SettingWithCopyWarning`-pandas warning - not sure if there is any other way..

Reply:
But in general I'm not sure if this PR should modify any files outside of `experimental` (except for the docstring maybe)..?

Reply:
I double checked, and this (and also the L513/L520 change) were to get rid of this `SettingWithCopyWarning`. Just let me know if I should roll these changes back or not :)

Reply:
I can't see the suggested change somehow.. can you post it again?

however when I checked the L520 change below for causing the warning, I also checked this part here - and it seems to be needed to avoid the warning:

```
high_var = np.zeros(df.shape[0])
high_var[:n_top_genes] = True
df['highly_variable'] = high_var.astype(bool)
df = df.loc[adata.var_names, :]
```

does not cause a warning, while

```
df['highly_variable'] = False
df.highly_variable.iloc[:n_top_genes] = True
df = df.loc[adata.var_names]
```


(I'm using this piece of code to test that:

```
import scanpy as sc
from scipy.sparse import csr_matrix

dtype = 'float32'
sparsity_func = csr_matrix
from scanpy.tests.helpers import _prepare_pbmc_testdata
adata = _prepare_pbmc_testdata(sparsity_func, dtype, small=True)


sc.pp.normalize_total(adata)
sc.pp.log1p(adata)
adata.obs.batch = adata.obs.batch.astype("category")
sc.pp.highly_variable_genes(adata,flavor='cell_ranger',batch_key='batch',n_top_genes=100)
```


So I left it unchanged for now. What was your suggested alternative?